### DISCUSSION TITLE
Any assembler project from scratch for ARM Cortex tutorial ?

### DISCUSSION THREAD

**Discussion by LAPSyLAB** [eclipse-embed-cdt/eclipse-plugins#539] [Project templates plug-ins Q&A]
Hello,

I'd like to create minimal assembler project for ARM Cortex M4 and M7 CPUs. 
I can only find templates or tutorials for C projects under Eclipse, but couldn't find 
tutorial about creating minimal assembler projects.

Any tutorial I could read/study ?

Thanks in advance,
regards.

**Comment by ilg-ul** [comment]
Sorry, but I don't know of templates or tutorials for assembler projects, and generally I do not encourage people of writing assembler code, since modern compilers usually generate better code.

**Reply by LAPSyLAB** [reply]
I agree, but I'm using assembly for instruction on Computer Architecture...

**Reply by TommyMurphyTM1234** [reply]
I'm not sure that Eclipse Embedded CDT is the best way to do that.
In fact, remember that a key selling point of Cortex-M was that, because the initial stack pointer was stored at offset 0 and the initial PC at offset 4, this meant that software could be written completely in a C without the need for any assembly code at all. In practice this may not always be the case, but it is possible.

I suspect that you need to reconsider what platform and tools might best suit your needs for learning (or teaching?) about Cortex-M assembly code/coding/architecture.

I'm pretty sure that there are many existing resources for doing that sort of thing. I would start with the Arm Developer site/resources.

* https://developer.arm.com/ip-products/processors/cortex-m/

**Comment by eraus** [comment]
> Hello,
> 
> I'd like to create minimal assembler project for ARM Cortex M4 and M7 CPUs. I can only find templates or tutorials for C projects under Eclipse, but couldn't find tutorial about creating minimal assembler projects.
> 
> Any tutorial I could read/study ?
> 
> Thanks in advance, regards.

@LAPSyLAB, have you found a solution/template using the Eclipse Embedded CDT for assembly programming? I have used Keil for mixed C and assembly programming, which works well. Now I am thinking of moving to Eclipse Embedded CDT. I need to do this mixed programming as well. If you have found a solution, could you please shed some light?

Many thanks!
